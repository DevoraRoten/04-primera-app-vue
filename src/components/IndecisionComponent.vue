<template>
<div>
 <img alt="Vue logo" v-show="!url" src="../assets/logo.png">
 <img :src="url" v-show="url" alt="">
   <div class="bg-dark"></div>

   <div class="indecision-conainer">
      <input type="text" v-model="question" placeholder="Hazme una pregunta">
      <p>Recuerda  terminar con un signo de interrogacion (?)</p>
   </div>

   <div>
      <h2 v-show="isValidQuestion">{{question}}</h2>
      <h1 v-show="isValidQuestion">{{ answer}}</h1>
   </div>

</div>
  

</template>

<script>
export default {
   data(){
      return{
         question: null,
         answer: null,
         url: null,
         isValidQuestion: false
      }
      
   },
   methods: {
      async getAnswer(){
         this.answer= 'Pensando...'
         const {answer,image} = await fetch('https://yesno.wtf/api').then(r=>r.json())
         console.log(answer)
         this.answer=answer==="yes"? 'SÃ­!' : 'No' 
         this.url=image

         
      }
   },
   watch: {
      question(value){
         this.isValidQuestion=false
         if(!value.includes('?')) return
         this.isValidQuestion=true
         this.getAnswer()
      }

   }

}
</script>

<style scoped>

   

</style>