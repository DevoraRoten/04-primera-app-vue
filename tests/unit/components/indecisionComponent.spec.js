import { shallowMount } from "@vue/test-utils";
import IndecisionComponent from "@/components/IndecisionComponent"
describe('indecicion component', ()=>{

    let wrapper;
    beforeEach(()=>{
        wrapper = shallowMount(IndecisionComponent)
    })
    test('Debe hacer match con el snapshop',()=>{
        expect(wrapper.html()).toMatchSnapshot()
    })
})