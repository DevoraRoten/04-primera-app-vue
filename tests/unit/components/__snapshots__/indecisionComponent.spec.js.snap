// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indecicion component Debe hacer match con el snapshop 1`] = `
<div><img alt="Vue logo" src=""><img alt="" style="display: none;">
  <div class="bg-dark"></div>
  <div class="indecision-conainer"><input type="text" placeholder="Hazme una pregunta">
    <p>Recuerda terminar con un signo de interrogacion (?)</p>
  </div>
  <div>
    <h2 style="display: none;"></h2>
    <h1 style="display: none;"></h1>
  </div>
</div>
`;
